src/app/shared/services/overlay.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable, ComponentFactoryResolver } from '@angular/core';
+import { Overlay, OverlayConfig, OverlayRef } from '@angular/cdk/overlay';
+import { ComponentPortal } from '@angular/cdk/portal';
+import { WinnerAnnounceComponent } from 'app/winner-announce/winner-announce.component';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class OverlayService {
```

Comment:
Using the material cdk for overlays, in this case just for announcing the winner of the spin